05-JWT-Basics/starter/controllers/main.js

```diff
@@ -0,0 +1,45 @@
+//check username, password in post(login) request
+//if exist create new JWT
+//send back to front-end
+
+//setup authentication so only the request with JWT can access the dashboard
+const jwt = require("jsonwebtoken");
+const {BadRequestError} = require("../errors");
+
+const login = async (req, res) => {
+  const { username, password } = req.body;
+
+  //validate the values By:
+  //mongoose validation
+  //Joi
+  //check in the controller
+
+  if (!username || !password) {
+    throw new BadRequestError("Please provide an email and password");
+  }
+  //just for demo, normally provided by DB!!
+  const id = new Date().getDate();
+
+  //try to keep payload small, better experience for user
+  // just for demo, in production use long, complex and unguessable string value!!!!!!!!!
+  const token = jwt.sign({ id, username }, process.env.JWT_SECRET, {
```

Comment:
Check that `JWT_SECRET` is defined at  startup, and fail early by throwing an error right away if it isn't.